tools/src/main/scala/scala/scalanative/optimizer/pass/StringLowering.scala

```diff
@@ -18,7 +18,7 @@ class StringLowering(implicit top: Top) extends Pass {
   private val stringFieldNames = {
     val node  = ClassRef.unapply(StringName).get
     val names = node.layout.entries.map(_.name)
-    assert(names.length == 4, "java.lang.String is expected to have 4 fields")
+    assert(names.length == 5, "java.lang.String is expected to have 4 fields")
```

Comment:
The message should be updated too: "[...] **5** fields"

Reply:
@tindzk Nice catch! I fixed it in the latest.